src/basic/cgroup-util.c

```diff
@@ -375,6 +404,15 @@ int cg_kill_recursive(
         assert(path);
         assert(sig >= 0);
 
+        if (cg_kill_supported() && sig == SIGKILL &&
+            FLAGS_SET(flags, CGROUP_REMOVE) && !FLAGS_SET(flags, CGROUP_IGNORE_SELF)) {
+                r = cg_kill_kernel(path);
```

Comment:
just `return cg_kill_kernel(...` here, given there's no logging or anything

Reply:
or even better, given the check is embedded, something like:

```
        if (sig == SIGKILL && FLAGS_SET(flags, CGROUP_REMOVE) && !FLAGS_SET(flags, CGROUP_IGNORE_SELF)) {
                r = cg_kill_kernel(path);
                if (r != -EOPNOTSUPP)
                        return r;
```

Reply:
`path` here won't be the right thing, no? You need to pass `cgroup.kill` too?

Reply:
Yes, good catch. Should I use `cg_get_path` or does `path_join(path, "cgroup.kill")` suffice?

Reply:
`cg_get_path(controller, path, "cgroup.kill"`,